### PR TITLE
Api documentation

### PR DESCRIPTION
**Pull request by wolterbwambale**
### Project requirements

***NOTE:*** _If you transitioned from 1.0 Microverse program use the [Social Media website](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/) you have built._

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well

### CODE REVIEW

File: app/controllers/api/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Api::PostsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @posts = Post.all
+    render json: @posts
+  end
+end
```
Comment (shella12) [inline app/controllers/api/posts_controller.rb]: - [ ] Great job on creating the API for your posts however you need to make sure that when I try to access the URL with a specific user_is like `/api/users/1/posts` it only lists posts from user 1 and not all the posts that exist in the database

File: app/controllers/api/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Api::CommentsController < ApplicationController
+  load_and_authorize_resource
+  def index
+    @comments = Comment.all
+    render json: @comments
+  end
```
Comment (shella12) [inline app/controllers/api/comments_controller.rb]: - [ ] Please make sure when I try to access comments of a specific post like for example `/users/1/posts/1/comments` API endpoints only lists the comments of post 1 by user 1 and not all the comments in the database.
- [ ] [OPTIONAL] Please fix the `initialize` method in the ability.rb model as its causing an error which is hindering the smooth running of the API testing

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @wolterbwambale and @henokkhm, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- All linters passed ✅ 
- Added all API endpoints ✅   
- Created routes for API endpoints ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @wolterbwambale and @henokkhm  👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ required changes are implemented
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**